rskj-core/src/main/java/co/rsk/Start.java

```diff
@@ -38,4 +45,45 @@ public static void main(String[] args) {
             System.exit(1);
         }
     }
+
+    // this feature is only needed until the secondFork (TBD) network upgrade is activated.
+    // the whole method should be deleted after that.
+    private static void migrateStateToUnitrieIfNeeded(RskContext ctx) {
+        String databaseDir = ctx.getRskSystemProperties().databaseDir();
+        // we need to check these before the data sources are init'ed
+        boolean hasUnitrieState = Files.exists(Paths.get(databaseDir, "unitrie"));
+        boolean hasOldState = Files.exists(Paths.get(databaseDir, "state"));
+        if (hasUnitrieState) {
+            // the node has been migrated
+            return;
+        }
+
+        if (!hasOldState) {
+            // node first start or after reset
+            return;
+        }
+
+        // this block number has to be validated before the release to ensure the migration works fine for every user
+        long minimumBlockNumberToMigrate = ctx.getRskSystemProperties().getDatabaseMigrationMinimumHeight();
+        Block blockToMigrate = ctx.getBlockStore().getBestBlock();
+        if (blockToMigrate == null || blockToMigrate.getNumber() < minimumBlockNumberToMigrate) {
+            logger.error(
+                    "The database can't be migrated because the node wasn't up to date before upgrading. " +
+                    "Please reset the database or sync past block {} with the previous version to continue.",
+                    minimumBlockNumberToMigrate
+            );
+            logger.error("Reset database or continue syncing with previous version");
+            System.exit(1);
```

Comment:
Let's remember to remove the `unitrie` folder in case of error

Reply:
done!